libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -172,8 +171,8 @@ internal class RouteAlternativesController constructor(
         nativeAlternatives: List<RouteAlternative>,
         block: (List<NavigationRoute>, RouterOrigin) -> Unit,
     ) {
-        val alternatives: List<NavigationRoute> = runBlocking {
-            nativeAlternatives.mapIndexedNotNull { index, routeAlternative ->
+        mainJobControl.scope.launch {
```

Comment:
maybe we should cancel the Job if `processRouteAlternatives` is called faster then previous completes?

Reply:
or cancel when user do `MapboxNavigation.setRoute(emptyList())`

Reply:
Good catch :+1:

Reply:
>  a new callback probably shouldn't be dispatched if the previous one has not returned yet 

If it is about RouterParser callback(not clear for me, because it was a comment for previous version of code), then yes - we schedule it to the same run loop from which parsing method was called, i.e. callbacks should come one by one.

Reply:
This thread is about `RouteAlternativesObserver` calls.